### PR TITLE
Fix `sproxyd_timeout` not being cast as `float`

### LINKED ISSUE #102: Setting a custom `sproxyd_conn_timeout` doesn't work in master
If one has the following object-server.conf: ```sproxyd_proxy_timeout = 20.0``` then swift-object will throw exception:
```
body=body, headers=headers)#012  File "/usr/local/lib/python2.7/dist-packages/urllib3/connectionpool.py", line 367, in _make_request#012
conn.sock.settimeout(read_timeout)#012  File "/usr/lib/python2.7/dist-packages/eventlet/greenio.py", line 381, in settimeout#012
raise TypeError('a float is required)#012TypeError: a float is required (txn: txef1445ed13eb4296bed66-00552bc206)
```

Because the urllib3 library expects timeouts value to be `float` and we pass `str`

This is a regression compared to branch 0.3.

### PR DESCRIPTION
**Pull request by JordanP**
Fixes #102

### CODE REVIEW

File: swift_scality_backend/server.py [hunk 1]
```diff
@@ -48,10 +48,20 @@ def setup(self, conf):
...
-        self._filesystem = SproxydClient(sproxyd_urls,
-                                         conf.get('sproxyd_conn_timeout'),
-                                         conf.get('sproxyd_proxy_timeout'),
-                                         self.logger)
+        # We can't pass `None` as value for sproxyd_*_timeout because it will
+        # override the defaults set in SproxydClient
+        kwarg = {}
+        try:
+            kwarg['sproxyd_conn_timeout'] = float(conf.get('sproxyd_conn_timeout'))
+        except TypeError:
```
Comment (NicolasT) [inline swift_scality_backend/server.py]: I guess a `TypeError` is only raised when `conf.get` returns `None`, but I'd be a bit more specific then, something along the lines of

```python
sproxyd_conn_timeout = conf.get('sproxyd_conn_timeout')
if sproxyd_conn_timeout is not None:
    kwargs['sproxyd_conn_timeout'] = float(sproxyd_conn_timeout)
```
and similar for `sproxyd_proxy_timeout`, or even go further:

```python
def get_config(name, fun):
    value = conf.get(name)
    return fun(value) if value is not None else None

kwargs = {
    'sproxyd_conn_timeout': get_config('sproxyd_conn_timeout', float),
    'sproxyd_proxy_timeout': get_config('sproxyd_proxy_timeout', float),
}
```

**Comment by JordanP** [comment]
I like this approach ! Thanks.

**Comment by JordanP** [comment]
In fact the 2nd approach doesn"t work. As I stated in the code comment
```
+        # We can't pass `None` as value for sproxyd_*_timeout because it will
+        # override the defaults set in SproxydClient
```

**Comment by NicolasT** [comment]
LGTM

**Pull request merged**